app/routes/docs.tsx

```diff
@@ -13,13 +13,16 @@ import parseRemixContext from "~/data/parseRemixContext.server";
 import getMysql from "~/data/mysql.server";
 import { apps } from "data/schema";
 import { ne } from "drizzle-orm/expressions";
+import path from "path";
 
 const DirectoryLink = ({
   level = 0,
   ...d
 }: DirectoryNode & { level?: number }) => {
   const matches = useMatches();
-  const pathSelected = matches.slice(-1)[0].pathname;
+  const [pathSelected, setPathSelected] = useState(
+    path.normalize(matches.slice(-1)[0].pathname)
+  );
```

Comment:
`pathSelected` should always be derived, we shouldn't need to manage it with state. let's nix the `onClick` and then run the replace that we have below here:

```typescript
const pathSelected = matches.slice(-1)[0].pathname.replace(/^[\\/]?docs[\\/]/, "");
```

Notice that I also added a `?` to the regex to make the leading path optional